server/welcomebot.go

```diff
@@ -228,12 +230,55 @@ func (p *Plugin) processActionMessage(messageTemplate MessageTemplate, action *A
 }
 
 func (p *Plugin) joinChannel(action *Action, channelName string) {
-	if channel, err := p.API.GetChannelByName(action.Context.TeamID, channelName, false); err == nil {
-		if _, err := p.API.AddChannelMember(channel.Id, action.Context.UserID); err != nil {
-			p.API.LogError("Couldn't add user to the channel, continuing to next channel", "user_id", action.Context.UserID, "channel_id", channel.Id)
+	// If it begins with @ create a DM channel
+	if strings.HasPrefix(channelName, "@") {
+		r := []rune(channelName)
+		dmUser, userErr := p.API.GetUserByUsername(string(r[1:]))
```

Comment:
You can perform the same slicing on strings as you can with arrays:

```suggestion
		dmUser, userErr := p.API.GetUserByUsername(channelName[1:])
```